lib/rust/ensogl/component/list-editor/src/lib.rs

```diff
@@ -1077,8 +1082,9 @@ impl<T: display::Object + CloneRef + 'static> Model<T> {
                 warn!("An element was inserted without a placeholder. This should not happen.");
                 index
             };
+            let index = self.item_or_placeholder_index_to_index(actual_index);
             self.reposition_items();
-            self.item_or_placeholder_index_to_index(actual_index)
+            index
```

Comment:
Does this change does any logic change or it was just syntactic one?

Reply:
It does change the logic. We have to reposition the items after calculating the item index. Repositioning will often delete collapsed placeholders, which made the index conversion incorrect and caused panics down the line.

Reply:
oh!